From 39417cdb0db6b80b6a8fd38e05b77e75d90717b8 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 20 Dec 2011 14:19:37 +0100
Subject: [PATCH] Do not import TCP_DENIED lines

Those pages were never showed to the client so there is no point in store
those for analyzing. Some applications (like Nokia connection suite) will
ignore an auth reject and will retry several times per second which will
flood the logs.
---
 weblog-model.lua |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/weblog-model.lua b/weblog-model.lua
index 07222fd..fe6fc37 100644
--- a/weblog-model.lua
+++ b/weblog-model.lua
@@ -652,6 +652,12 @@ local function parsesquidlog(line)
 
 	checkwords(logentry)
 
+	-- Don't care about TCP_DENIED so apps like dropbox, nokia connection
+	-- suite does not flood our logs.
+	if logentry.code == nil or logentry.code == "TCP_DENIED" then
+		return nil
+	end
+
 	-- Don't care about local requests (from DG) (this check also removes blank lines)
 	if logentry.clientip and logentry.clientip ~= "127.0.0.1" then
 		logentry.logdatetime = os.date("%Y-%m-%d %H:%M:%S", logentry.logdatetime)..string.match(logentry.logdatetime, "%..*")
-- 
1.7.8.3

